apps/api/src/services/db/statsService.ts

```diff
@@ -202,40 +202,42 @@ export const getProviderGraphData = async (dataName: ProviderStatsKey) => {
 
   const getter = (block: ProviderStats) => (typeof block[dataName] === "number" ? (block[dataName] as number) : parseInt(block[dataName] as string) || 0);
 
-  const result = await cacheResponse(
-    60 * 5, // 5 minutes
-    cacheKeys.getProviderGraphData,
-    async () => {
-      return await chainDb.query<ProviderStats>(
-        `SELECT d."date", (SUM("activeCPU") + SUM("pendingCPU") + SUM("availableCPU")) AS "cpu", (SUM("activeGPU") + SUM("pendingGPU") + SUM("availableGPU")) AS "gpu", (SUM("activeMemory") + SUM("pendingMemory") + SUM("availableMemory")) AS memory, (SUM("activeStorage") + SUM("pendingStorage") + SUM("availableStorage")) as storage, COUNT(*) as count
-         FROM "day" d
-         INNER JOIN (
-            SELECT DISTINCT ON("hostUri",DATE("checkDate")) 
-              DATE("checkDate") AS date, 
-              ps."activeCPU", ps."pendingCPU", ps."availableCPU", 
-              ps."activeGPU", ps."pendingGPU", ps."availableGPU", 
-              ps."activeMemory", ps."pendingMemory", ps."availableMemory", 
-              ps."activeEphemeralStorage" + COALESCE(ps."activePersistentStorage", 0) AS "activeStorage", ps."pendingEphemeralStorage" + COALESCE(ps."pendingPersistentStorage", 0) AS "pendingStorage", ps."availableEphemeralStorage" + COALESCE(ps."availablePersistentStorage", 0) AS "availableStorage", 
+  const result = removeLastIfZeroCount(
+    await cacheResponse(
+      60 * 5, // 5 minutes
```

Comment:
non-blocking: a constant from `date-fns` could be used - no commend would be needed